ssl/rio/rio_notifier.c

```diff
@@ -71,9 +71,16 @@ static int create_socket(int domain, int socktype, int protocol)
      * Use WSASocketA to create a socket which is immediately marked as
      * non-inheritable, avoiding race conditions if another thread is about to
      * call CreateProcess.
+     * NOTE: windows xp (0x501) doesn't support the non-inheritance flag here
+     * but preventing inheritance isn't mandatory, just a safety precaution
+     * so we can get away with not including it for older platforms
      */
+#  if _WIN32_WINNT > 0x501
```

Comment:
This basically means we are not going to set this by default because by default we define _WIN32_WINNT to be 0x0501.

Do we perhaps want to change the default now? People can override in their builds.

Or could we do a runtime check?

Reply:
Hmm... with this kind of runtime check you could just change this `# if` to `#  ifdef WSA_FLAG_NO_HANDLE_INHERIT`